lib/rucio/common/schema/__init__.py

```diff
@@ -140,10 +142,16 @@ def load_schema_for_vo(vo: str) -> None:
 def validate_schema(name: str, obj: Any, vo: str = 'def') -> None:
     if vo not in schema_modules:
         load_schema_for_vo(vo)
-    if not hasattr(schema_modules[vo], 'validate_schema'):
-        _get_generic_schema_module().validate_schema(name, obj)
-        return
-    schema_modules[vo].validate_schema(name, obj)
+    if hasattr(schema_modules[vo], 'SCHEMAS') and name in schema_modules[vo].SCHEMAS:
+        schema = schema_modules[vo].SCHEMAS.get(name, {})
+    else:
+        # if schema not available in VO module, fall back to generic module
+        schema = _get_generic_schema_module().SCHEMAS.get(name, {})
+    try:
+        if obj:
+            validate(obj, schema)
+    except ValidationError as error:  # NOQA, pylint: disable=W0612
+        raise exception.InvalidObject(f'Problem validating {name}: {error}')
```

Comment:
Validation only happens `if obj` is `True`, so I think it could be more optimal to have this `if obj` check be done as the first thing in the function, so that we avoid loading the schema and preparing for validation if we don't have `obj`